packages/clerk-js/src/ui/hooks/useFetch.ts

```diff
@@ -77,6 +77,7 @@ export const useFetch = <K, T>(
   const cached = useSyncExternalStore(subscribeCache, getCache);
 
   useEffect(() => {
+    fetcherRef.current = fetcher; // Only update fetcher when params change
```

Comment:
Adding this after pairing with @panteliselef 
Next steps I'm planning to add some tests for `useFetch` in the main and then I will add this again